lxd/instance/drivers/driver_lxc.go

```diff
@@ -1966,6 +1966,13 @@ func (d *lxc) startCommon() (string, []func() error, error) {
 		volatileSet["volatile.uuid"] = instUUID
 	}
 
+	// For a container instance, we must also set the generation UUID.
+	genUUID := d.localConfig["volatile.uuid.generation"]
+	if genUUID != "" {
```

Comment:
This reads to me as "if the instance already has a generation ID then on start, replace it with the instance UUID" which seems wrong to me.

Reply:
@tomponline I updated it to 
```go
// For a container instance, we must also set the generation UUID.
genUUID := d.localConfig["volatile.uuid.generation"]
if genUUID == "" {
	genUUID = instUUID
	volatileSet["volatile.uuid.generation"] = genUUID
}
```